lib/rspec/buildkite/analytics/session.rb

```diff
@@ -52,9 +52,22 @@ def initialize(url, authorization_header, channel)
 
       @logger = Logger.new
 
-      connect
-    rescue TimeoutError, InitialConnectionFailure => e
-      $stderr.puts "rspec-buildkite-analytics could not establish an initial connection with Buildkite due to #{e.message}. You may be missing some data for this test suite, please contact support."
+      reconnection_count = 0
+
+      begin
+        reconnection_count += 1
+        connect
+      rescue TimeoutError, InitialConnectionFailure => e
+        @logger.write("failed initial connection attempt #{reconnection_count} due to #{e}")
```

Comment:
I think if this code gets restructured a bit we can provide some better debug messages like the one here and in my other comments.

```suggestion
        @logger.write("rspec-buildkite-analytics could not establish an initial connection with Buildkite due to #{e}. Attempting retry #{reconnection_count} of #{MAX_RECONNECTION_ATTEMPTS}...")
```